lib/Runtime/Base/ThreadContext.cpp

```diff
@@ -910,19 +911,19 @@ ThreadContext::IsNumericProperty(Js::PropertyId propertyId)
 const Js::PropertyRecord *
 ThreadContext::FindPropertyRecord(const char16 * propertyName, int propertyNameLength)
 {
-    Js::PropertyRecord const * propertyRecord = nullptr;
-
-    if (IsDirectPropertyName(propertyName, propertyNameLength))
-    {
-        propertyRecord = propertyNamesDirect[propertyName[0]];
-        Assert(propertyRecord == propertyMap->LookupWithKey(Js::HashedCharacterBuffer<char16>(propertyName, propertyNameLength)));
-    }
-    else
+    if (propertyNameLength < 2)
     {
-        propertyRecord = propertyMap->LookupWithKey(Js::HashedCharacterBuffer<char16>(propertyName, propertyNameLength));
+        if (propertyNameLength == 0) return this->GetEmptyStringPropertyRecord();
```

Comment:
>if (propertyNameLength == 0) return this->GetEmptyStringPropertyRecord(); [](start = 8, length = 73)

move this line out? and use {} around the return.

and maybe still keep propertyRecord for all code path, so in chk build it's easier for debugging, and in free build it will by optimized by the compiler anyway